substrate/frame/nomination-pools/src/mock.rs

```diff
@@ -423,6 +423,61 @@ pub fn fully_unbond_permissioned(member: AccountId) -> DispatchResult {
 	Pools::unbond(RuntimeOrigin::signed(member), member, points)
 }
 
+pub fn pending_rewards_for_pool(pool: PoolId) -> Balance {
+	let bonded_pool = BondedPools::<T>::get(pool).unwrap();
+	let reward_pool = RewardPools::<T>::get(pool).unwrap();
+
+	let current_rc = if !bonded_pool.points.is_zero() {
+		let commission = bonded_pool.commission.current();
+		reward_pool
+			.current_reward_counter(pool, bonded_pool.points, commission)
+			.unwrap()
+			.0
+	} else {
+		Default::default()
+	};
+
+	PoolMembers::<T>::iter()
```

Comment:
We can't really do this as this on-chain I guess since this is an unbounded number. Not sure what's the alternative? Some way to page it. We can let any caller just top up any amount and if it is greater than deficit, those tokens will be stuck in reward pool forever and would be equivalent to burning money.